Fix/update training booked spot!!

ebuka-nwabueze opened the pull request:
- Added certificate download feature.
- Update the Account page for booked trainings.

ebuka-nwabueze committed:
- Add new apis.
- update feature card component with booked spot.

Ozoemenamsolomon committed:
Merge branch 'main' into fix/update-traning-booked-spot

Ozoemenamsolomon committed:
Fix build

ebuka-nwabueze committed:
update the api url

remove api call in server env to direct DB access

ebuka-nwabueze committed:
update the training days to add 1 to the difference calculation and update component

ebuka-nwabueze committed:
update infobox for each training

ebuka-nwabueze committed:
remove console log

ebuka-nwabueze committed:
update infobox text

ebuka-nwabueze committed:
add trainings to account page

update training order component

ebuka-nwabueze committed:
add verify training ticket page

ebuka-nwabueze committed:
add verify training ticket api

ebuka-nwabueze committed:
add confirmation api and logic to component

ebuka-nwabueze committed:
- complete the training ticket verification
- Update Training Order component css

ebuka-nwabueze committed:
added the download certificate api and component

ebuka-nwabueze committed:
remove commented code

ebuka-nwabueze committed:
remove console.log

ebuka-nwabueze committed:
add the paint images in the certificate

ebuka-nwabueze committed:
update font-family

Ozoemenamsolomon commented on src/pages/docs/training-certificate/[orderId]/[participantId].tsx:
```diff
@@ -0,0 +1,282 @@
+	const orderId = router.query.orderId as string;
+	const participantId = router.query.participantId as string;
+
+	useEffect(() => {
```
do this in the `getServerProps` function. That's the benefit of using next js over ordinary react

ebuka-nwabueze replied:
Done ✅✅

Ozoemenamsolomon commented on src/pages/docs/training-certificate/[orderId]/[participantId].tsx:
```diff
@@ -0,0 +1,282 @@
+		});
+	};
+
+	if (!user)
```
look at other authenticated routes, we use `withPageAuthRequired` provided by auth0 to make sure only authenticated users get to the page at all

ebuka-nwabueze replied:
Done ✅✅

Ozoemenamsolomon commented on src/pages/api/update-training-bookedspot.ts:
just create a `training-spot` route and use the REST methods respectively: in this case, use `PUT`

ebuka-nwabueze replied:
Done ✅✅

Ozoemenamsolomon commented on src/pages/api/download-training-certificate.ts:
the route name could be improved, the route is just providing data for the certificate that would be processed in the frontend

ebuka-nwabueze replied:
Deleted the api route due to accessing the database directly from getServerSideProps.

ebuka-nwabueze committed:
change authentication in the training attendance data fetching in serverside props

delete download training api

update training booked spot api to use PUT method and naming convenntion

Ozoemenamsolomon committed:
Merge branch 'main' into fix/update-traning-booked-spot
